include/fmt/format.h

```diff
@@ -1871,22 +1909,77 @@ auto write_float(OutputIt out, const DecimalFP& fp,
   });
 }
 
+template <typename T, FMT_ENABLE_IF(std::is_floating_point<T>::value)>
+FMT_CONSTEXPR20 bool isinf(T value) {
+  if (is_constant_evaluated()) {
+#if defined(__cpp_lib_bit_cast) && defined(__cpp_if_constexpr)
+    if constexpr (is_fast_float<T>::value) {
+      using floaty =
+          std::conditional_t<std::is_same<T, long double>::value, double, T>;
```

Comment:
Let's move this to a new `float_type` trait similar to https://github.com/fmtlib/fmt/blob/117fc67077ed3bbd10ad4ae83f934d49e5da2cec/include/fmt/core.h#L1231 to reduce copy-paste.

Reply:
I simplified `isinf` and `isfinite`, applying the same ideas as for `signbit`, so it's probably irrelevant now.